core/dbt/clients/agate_helper.py

```diff
@@ -33,11 +34,21 @@ def table_from_data(data, column_names):
         return table.select(column_names)
 
 
-def table_from_data_explicit(data, column_names):
+def table_from_data_flat(data, column_names):
     "Convert list of dictionaries into an Agate table"
 
-    rows = [list(r.values()) for r in data]
-    return agate.Table(rows, column_names)
+    rows = []
+    for _row in data:
+        row = []
+        for value in list(_row.values()):
+            if isinstance(value, (dict, list, tuple)):
+                row.append(json.dumps(value))
+            else:
+                row.append(value)
+        rows.append(row)
+
+    table = agate.Table(rows, column_names)
+    return table.select(column_names)
```

Comment:
why the `.select`? Doesn't passing the column names to the table constructor make this redundant?